libraries/AP_Param/AP_Param.cpp

```diff
@@ -1096,7 +1096,9 @@ void AP_Param::save(bool force_save)
         // when we are disarmed then loop waiting for a slot to become
         // available. This guarantees completion for large parameter
         // set loads
+        hal.scheduler->expect_delay_ms(500);
```

Comment:
Unit mixup.

I suspect this kind of thing is going to be so common, that unless we wrap this internally into the delay() call in the HAL with a check if we are on the main thread, this is going to be a constant problem everywhere. Given the relative rarity of calling delay() I think it's worth the overhead.